packages/vitest/src/node/workspace/resolveWorkspace.ts

```diff
@@ -311,8 +318,7 @@ function cloneConfig(project: TestProject, { browser, ...config }: BrowserInstan
       testerHtmlPath: testerHtmlPath ?? currentConfig.testerHtmlPath,
       screenshotDirectory: screenshotDirectory ?? currentConfig.screenshotDirectory,
       screenshotFailures: screenshotFailures ?? currentConfig.screenshotFailures,
-      // TODO: test that CLI arg is preferred over the local config
-      headless: project.vitest._options?.browser?.headless ?? headless ?? currentConfig.headless,
+      headless: headless ?? currentConfig.headless,
```

Comment:
this doesn't seem to be related

Reply:
It is related.

Reply:
I don't see how it is related. Wouldn't this actually break the use case and you won't be able to override the headless mode?